Android/app/src/main/java/app/intra/util/DnsQueryTracker.java

```diff
@@ -124,7 +124,7 @@ public void recordTransaction(DnsTransaction transaction) {
     }
   }
 
-  public void sync() {
+  public synchronized  void sync() {
```

Comment:
Likewise, not sure why you need to have two different methods of synchronizing?

Reply:
Oops.  I started doing this cleanup before and then decided I couldn't justify it in a UI performance PR, but forgot to revert this bit.  Luckily, now that the reviewer has complained it's totally justified.